Source/Foundation/NSObject.swift

```diff
@@ -48,7 +54,9 @@ extension NSObject {
     public func rex_classProperty<T: AnyObject>(keyPath: StaticString, placeholder: () -> T) -> MutableProperty<T> {
         return associatedProperty(self, keyPath: keyPath, placeholder: placeholder)
     }
+}
 
+extension NSObjectType where Self: NSObject, InstanceType == Self {
```

Comment:
> now require declaring `InstanceType` in each extension.

Yep, that is really what I'm concerned about. The same functionality is easily achieved by type inference if we use `associatedProperty` directly. If you agree, I'm willing to remove the series of `rex_*Property`.